### DISCUSSION TITLE
Allow user to choose directory for additional media

### DISCUSSION THREAD

**Discussion by e-Feitosa** [sircharlo/meeting-media-manager#2782] [Ideas] [answered]
### Is your idea related to a problem?

The idea is related to the discontinuation of WebDAV in M3

### Describe your idea

Include in the M3’s advanced settings / optional to detect media in a local folder.
For example, when setting up Dropbox/OneDrive/Google Drive on KH and pointing this folder to on M3, upon starting the program, a warning will be displayed indicating that there is additional media file and whether you want to import into the program.

### Describe alternatives you've considered

If it were possible to include this advanced option, it would solve the problem we have with WebDAV.

**Comment by e-Feitosa** [comment]
@sircharlo and @mtdvlpr 
 Is this option (_**Allow user to choose directory for additional media**_) on the task list related to the topic mentioned in this post (**_Needed refactors to eventually allow the additional media folder to be 'portable' or synced_**)?

https://github.com/sircharlo/meeting-media-manager/issues/2781#:~:text=Allow%20user%20to%20choose%20directory%20for%20additional%20media

**Reply by mtdvlpr** [reply]
Yes, it's present on the list:

> - [ ] Allow user to choose directory for additional media

**Reply by e-Feitosa** [reply]
That’s very good. We are in great need of this feature. I am looking forward to this feature being implemented.
_Thank you for your valuable work._

**Reply by advenimus** [reply]
Just a suggestion @sircharlo, but inside this custom media folder could live date structured directories (ie. 2024-10-30) and M3 could look for those inside this folder and auto import that media for that given media? This would replace the now deprecated webDAV functionality?

**Reply by e-Feitosa** [reply]
Excellent question [suggestion], if the M3, search in the folder this way, will solve all the problems we have without WebDAV, for example, for the meeting day 20241102, all files corresponding to the meeting that day would be imported into the M3:
20241102 - Image 1 
20241102 - Image 2
20241102 - Image 3
20241102 - Video 1

M3 would search the folder according to the mid-week and weekend meeting date. That would be great.

**Comment by sircharlo** [comment]
I'm going to tackle this this month.

**Reply by e-Feitosa** [reply]
Thanks so much for handling this! It means a lot.

**Comment by mtdvlpr** [comment, marked as answer]
Good news! This feature has been merged and will be in the next release :)

**Reply by advenimus** [reply]
@mtdvlpr - Awesome work brothers!

Would it be difficult to build in the future a functionality to save the queue file to this folder and preferences to a custom folder which can be synced via a cloud folder in a similar way that the old sync worked? That way cloud sync of preferences and the order of media can still be managed remotely for the brothers and then it's not on you to support any sync protocol, just an observed folder...

**Reply by sircharlo** [reply]
What do you mean by the queue file?

**Reply by advenimus** [reply]
Like the file that stores the order for the meeting queue that meeting. 

My idea (concept) is that the program can monitor that custom folder for numbered files (or some sort of naming format, folder structure, something like that) which would corespondent to the order to display them (similar to the old version) that way the correct order is display automatically upon observation of that folder. 

And similarly with the preferences file, point to a custom folder for easy sync of congregation settings, etc. 

Does that sound like a lot of work?

**Reply by mtdvlpr** [reply]
Continued in #3331
